pkg/scheduler/framework/session_plugins.go

```diff
@@ -317,6 +321,41 @@ func (ssn *Session) IsTaskAllocationOnNodeOverCapacityFn(task *pod_info.PodInfo,
 	}
 }
 
+func (ssn *Session) SubsetNodesFn(podGroup *podgroup_info.PodGroupInfo, isRealAllocation bool, initNodeSet node_info.NodeSet) ([]node_info.NodeSet, error) {
+	nodeSets := []node_info.NodeSet{initNodeSet}
+	for _, subsetNodesFn := range ssn.SubsetNodesFns {
+		log.InfraLogger.V(7).Infof(
+			"Running plugin func <%v> on podGroup <%s/%s>", subsetNodesFn, podGroup.Namespace, podGroup.Namespace)
+		newNodeSets := []node_info.NodeSet{}
+		for _, nodeSet := range nodeSets {
+			nodeSubsets, err := subsetNodesFn(podGroup, isRealAllocation, nodeSet)
+			if err != nil {
+				log.InfraLogger.V(6).Infof(
+					"Failed to run SubsetNodes function <%v> on job <%s/%s>", subsetNodesFn, podGroup.Namespace, podGroup.Namespace)
+				return nil, err
```

Comment:
As an infra function, I don't think it should log here - I suggest to return a meaningful error and let the caller log it instead.